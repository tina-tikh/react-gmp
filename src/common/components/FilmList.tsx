import * as React from 'react';
import {ReactNode} from 'react';

import {Movie} from '../models/movie';
import FilmCard from './FilmCard';
import styled from '../styled';

type FilmListProps = {
    films: Movie[];
}

const List = styled.ul`
    list-style-type: none;
    margin: 2rem;
    padding: 0;
`;

const ListItem = styled.li`
    display: inline-block;
`;

class FilmList extends React.Component<FilmListProps> {
    render(): ReactNode {
        return <List>
            {this.props.films.map((film: Movie) => <ListItem key={film.id}>
                <FilmCard film={film}/>
            </ListItem>)}
        </List>
    }
}

export default FilmList;
