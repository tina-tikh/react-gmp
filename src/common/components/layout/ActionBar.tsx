import { styled } from '../../theme';

const ActionBar = styled.section`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 2rem 6rem;
  background-color: ${(props) => props.theme.colors.dark};
`;

export default ActionBar;
