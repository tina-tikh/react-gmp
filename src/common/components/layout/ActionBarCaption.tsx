import { styled } from '../../theme';

const ActionBarCaption = styled.span`
  font-size: 1.4rem;
  font-weight: bold;
`;

export default ActionBarCaption;
