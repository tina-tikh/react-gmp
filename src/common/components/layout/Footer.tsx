import * as React from 'react';

import { styled } from '../../theme';
import Logo from './Logo';

const FooterBlock = styled.footer`
  padding: 2.5rem;
  text-align: center;
  background-color: ${(props) => props.theme.colors.dark};
`;

class Footer extends React.PureComponent {
  render() {
    return (
      <FooterBlock>
        <Logo></Logo>
      </FooterBlock>
    );
  }
}

export default Footer;
