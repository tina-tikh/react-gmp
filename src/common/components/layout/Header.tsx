import * as React from 'react';

import { styled } from '../../theme';
import Logo from './Logo';

const HeaderBlock = styled.header`
  padding: 3rem 6rem;
  color: ${(props) => props.theme.colors.light};
  background-color: ${(props) => props.theme.colors.darkest};
`;

class Header extends React.PureComponent {
  render() {
    return (
      <HeaderBlock>
        <Logo></Logo>
        {this.props?.children}
      </HeaderBlock>
    );
  }
}

export default Header;
