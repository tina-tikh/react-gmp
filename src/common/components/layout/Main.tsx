import * as React from 'react';

import { styled } from '../../theme';

const Main = styled.main`
  flex: 1;
  color: ${(props) => props.theme.colors.light};
  background-color: ${(props) => props.theme.colors.darker};
`;

export default Main;
