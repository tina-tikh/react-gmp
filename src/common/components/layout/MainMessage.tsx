import { styled } from '../../theme';

const MainMessage = styled.h1`
  margin: 5rem;
  text-align: center;
  font-size: 4rem;
`;

export default MainMessage;
