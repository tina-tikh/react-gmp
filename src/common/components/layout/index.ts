import ActionBar from './ActionBar';
import ActionBarCaption from './ActionBarCaption';
import Footer from './Footer';
import Header from './Header';
import Logo from './Logo';
import Main from './Main';
import MainMessage from './MainMessage';
import Page from './Page';

export {
  ActionBar,
  ActionBarCaption,
  Footer,
  Header,
  Logo,
  Main,
  MainMessage,
  Page,
};
