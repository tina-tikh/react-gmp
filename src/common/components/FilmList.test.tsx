import * as React from 'react';
import { mount, shallow } from 'enzyme';

import FilmList from './FilmList';
import { Movie } from '../models';
import { Page } from './layout';

const films: Movie[] = [
  {
    id: 447365,
    title: 'Guardians of the Galaxy Vol. 3',
    tagline: '',
    vote_average: 0,
    vote_count: 9,
    release_date: '2020-05-01',
    poster_path:
      'https://image.tmdb.org/t/p/w500/ldoY4fTZkGISMidNw60GHoNdgP8.jpg',
    overview: "The third film based on Marvel's Guardians of the Galaxy.",
    budget: 0,
    revenue: 0,
    genres: ['Action', 'Adventure', 'Science Fiction'],
    runtime: null,
  },
  {
    id: 424785,
    title: 'Transformers 7',
    tagline: '',
    vote_average: 0,
    vote_count: 0,
    release_date: '2019-06-26',
    poster_path:
      'https://image.tmdb.org/t/p/w500/432BowXw7a4fWXSONxBaFKqvW4f.jpg',
    overview: 'Plot unknown.',
    budget: 0,
    revenue: 0,
    genres: ['Science Fiction', 'Action', 'Adventure'],
    runtime: null,
  },
];

jest.mock('react-router-dom');

describe('<FilmList />', () => {
  it('should be rendered', () => {
    const component = shallow(<FilmList films={[]} />);
    expect(component).toMatchSnapshot();
  });

  it('should render passed films', () => {
    const component = mount(
      <Page>
        <FilmList films={films} />
      </Page>
    );
    expect(component.find('li').length).toEqual(films.length);
  });
});
