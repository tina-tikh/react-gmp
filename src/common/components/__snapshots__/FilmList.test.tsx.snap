// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilmList /> should be rendered 1`] = `<Styled(ul) />`;
